README.md
# ES-Utils for Flutter

### A Flutter package for few UI components like confirmDialog, inputDialog, Buttons etc 

 
[![](https://img.shields.io/pub/v/es_utils.svg)](https://pub.dartlang.org/packages/es_utils)
[![Flutter](https://github.com/ajumalp/flutter_es_utils/actions/workflows/flutter.yml/badge.svg?branch=main)](https://github.com/ajumalp/flutter_es_utils/actions/workflows/flutter.yml)


## Development tools required 
[![](https://img.shields.io/badge/Visual%20Studio%20Code-1.63-blue)](https://code.visualstudio.com/)

## Visual Studio Code Extensions used 
[![](https://img.shields.io/badge/Dart-2.12.0-blue)](https://marketplace.visualstudio.com/items?itemName=Dart-Code.dart-code)
[![](https://img.shields.io/badge/Flutter-2.8.0-blue)](https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter)

## Languages used 
![](https://img.shields.io/badge/Flutter-Dart-00979D)    

## Flutter Package Dependencies    
[![](https://img.shields.io/badge/URL_Launcher-v6.0.17-orange)](https://pub.dev/packages/url_launcher/versions/6.0.17) 


## Screenshots
<img src="https://raw.githubusercontent.com/ajumalp/flutter_es_utils/main/other/images/screenshots/scr-confirm.jpg" height="300">&nbsp;&nbsp;<img src="https://raw.githubusercontent.com/ajumalp/flutter_es_utils/main/other/images/screenshots/scr-input.jpg" height="300">&nbsp;&nbsp;<img src="https://raw.githubusercontent.com/ajumalp/flutter_es_utils/main/other/images/screenshots/scr-list.jpg" height="300">&nbsp;&nbsp;<img src="https://raw.githubusercontent.com/ajumalp/flutter_es_utils/main/other/images/screenshots/scr-main.jpg" height="300">&nbsp;&nbsp;<img src="https://raw.githubusercontent.com/ajumalp/flutter_es_utils/main/other/images/screenshots/scr-progress.jpg" height="300">
      
## Confirm Dialog 
```dart 
ESButton(
    'Show Confirm Dialog',
    onPressed: () => ESMessage.showConfirmDialog(
        context: context,
        title: 'Confirm Dialog',
        message: 'Are you sure?',
        buttonLeft: 'OK',
        buttonRight: 'Cancel',
    ),
)
```

## List Dialog 
```dart 
final List<bool> varList = [false, false, true, true, false];
ESButton(
    'List Dialog',
    onPressed: () => ESMessage.showListDialog(
        context: context,
        suffixText: 'Rows',
        selectedValue: 20,
        title: 'Row Count',
        multiSelectValues: varList,
        options: const [10, 20, 30, 50, 100],
    ),
)
```
## Input Dialog 
```dart
ESButton(
    'Show Input Dialog',
    onPressed: () => ESMessage.showInputDialog(
        context: context,
        title: 'Title',
        defaultValue: 'Default Value',
        hitText: 'Enter Value',
        onSubmit: (TextEditingController controller) => ESMessage.showInfoMessage(
        context,
        message: controller.text,
        ),
    ),
)
```


CHANGELOG.md
## [1.0.9] Release 1.0.9

* Added showModalDialog
* Added showListDialog
* Fixed many issues 

## [1.0.4] Release 1.0.4

* Added ESPlatform 
* Added ESLinkText 

## [1.0.3] Release 1.0.3

* Removed dart.io to make package support web platform also 
* Updated readme 

## [1.0.2] Release 1.0.2
 
* Added ESButton
* Added ESProgressIndicator
* Added ESMessage

## [1.0.1] - Initial Release

* Initial release


example/lib/main.dart
/*
 * Developed by Ajmal Muhammad P
 * Contact me @ support@erratums.com
 * https://erratums.com
 * Date created: 09-May-2021
 */

import 'package:es_utils/es_utils.dart';
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      debugShowCheckedModeBanner: false,
      theme: ThemeData(primarySwatch: Colors.blue),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final List<bool> varList = [false, false, true, true, false];
    return Scaffold(
      appBar: AppBar(
        title: const Text('ES-Utils Demo'),
      ),
      body: Padding(
        padding: EdgeInsets.symmetric(horizontal: ESPlatform.isSmallScreen(context) ? 50 : 30),
        child: Center(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
            children: <Widget>[
              ESButton(
                'Progress Dialog',
                onPressed: () => Navigator.push(
                  context,
                  MaterialPageRoute(
                    builder: (context) => FutureBuilder(
                      future: Future.delayed(const Duration(seconds: 2)),
                      builder: (context, AsyncSnapshot snapshot) {
                        if (snapshot.connectionState != ConnectionState.done) {
                          return ESMessage.showProgressIndicator(title: 'New Screen');
                        }
                        return Scaffold(
                          appBar: AppBar(title: const Text('New Screen')),
                          body: const Text('New Screen'),
                        );
                      },
                    ),
                  ),
                ),
              ),
              ESButton(
                'Show Input Dialog',
                onPressed: () => ESMessage.showInputDialog(
                  context: context,
                  title: 'Title',
                  defaultValue: 'Default Value',
                  hitText: 'Enter Value',
                  onSubmit: (TextEditingController controller) => ESMessage.showInfoMessage(
                    context,
                    message: controller.text,
                  ),
                ),
              ),
              ESButton(
                'Show Confirm Dialog',
                onPressed: () => ESMessage.showConfirmDialog(
                  context: context,
                  title: 'Confirm Dialog',
                  message: 'Are you sure?',
                  buttonLeft: 'OK',
                  buttonRight: 'Cancel',
                ),
              ),
              ESButton(
                'List Dialog',
                onPressed: () => ESMessage.showListDialog(
                  context: context,
                  suffixText: 'Rows',
                  selectedValue: 20,
                  title: 'Row Count',
                  multiSelectValues: varList,
                  options: const [10, 20, 30, 50, 100],
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
